### PR TITLE
FIX: Improvements to animated image pausing

### PR DESCRIPTION
**Pull request by pmusaraj**
- Adds pause/play icons in bottom/right corner
- Pauses animated images automatically on blur
- Resumes animated images when window focus returns
- Pauses on load when OS has `prefers-reduced-motion` enabled
- Adds subtle background color to `canvas` element, so that in slow connections where the full GIF loads slowly, pausing the animation will show something in stead of blank space

The icons are shown as follows: 

Pause icon is shown for non-touch devices on hover.
Play icon is shown on all devices when animation is paused.

### CODE REVIEW

**Review by eviltrout — APPROVED** [review]

File: app/assets/javascripts/discourse/app/initializers/animated-images.js [hunk 1]
```diff
@@ -8,29 +10,42 @@ export default {
...
+        if (opts.manualPause) {
+          img.classList.add("manually-paused");
         }
+        img.parentNode.classList.add("paused-animated-image");
+        img.parentNode.insertBefore(canvas, img);
+      }
+
+      function _resumeAnimation(img) {
+        img.previousSibling.remove();
+        img.parentNode.classList.remove("paused-animated-image");
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/initializers/animated-images.js:47]: Just a minor note, know that you can add/remove multiple classes at once:

```
div.classList.remove("foo", "bar", "baz");
```

https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

File: app/assets/javascripts/discourse/app/initializers/animated-images.js [hunk 2]
```diff
@@ -61,6 +96,33 @@ export default {
       });
 
       api.cleanupStream(_cleanUp);
+
+      // paused on load when prefers-reduced-motion is active, no need for blur/focus events
+      if (!prefersReducedMotion()) {
+        const images = "img.animated:not(.manually-paused)";
+
+        window.addEventListener("blur", () => {
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/initializers/animated-images.js:104]: we could probably remove these listeners in the `teardown()` hook of initializer for extra safety.

File: app/assets/stylesheets/common/base/topic-post.scss [hunk 3]
```diff
@@ -1235,15 +1235,59 @@ a.mention-group {
...
+      cursor: pointer;
+      display: none;
+    }
+  }
+
+  img.animated {
+    cursor: pointer;
+  }
+
+  html.no-touch
```
Comment (jjaffeux) [inline app/assets/stylesheets/common/base/topic-post.scss:1274]: Am I missing something here? Indentation seems weird for what seems like a top level selector.

Comment (pmusaraj) [reply app/assets/stylesheets/common/base/topic-post.scss:1274]: I suspect it's because the selector is quite long (this was Prettier's doing anyway).

**Pull request merged**